### PR TITLE
ocamlPackages.dune_2: 2.7.1 -> 2.8.1, ocamlPackages.mdx: 1.7.0 -> 1.8.0

### PR DESCRIPTION
**Pull request by anmonteiro**
###### Motivation for this change
https://github.com/ocaml/dune/releases/tag/2.8.0

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by anmonteiro** [comment]
This release includes https://github.com/ocaml/dune/pull/3973, which is especially important to support `ocaml-ng.ocamlPackages_4_12`

File: pkgs/development/tools/ocaml/dune/2.nix [hunk 1]
```diff
@@ -6,25 +6,17 @@ else
...
+    sha256 = "12yly2lp93ijhy7b72p6y2q3cr3yy3hk7rlmrh072py8a6d4s407";
   };
 
   buildInputs = [ ocaml findlib ];
 
   buildFlags = "release";
 
-  patches = [
-    # Fix setup.ml configure path. Remove with the next release.
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/ocaml/dune/2.nix:22]: Please remove fetchpatch from inputs.

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.mdx

**Review by vbgl — CHANGES_REQUESTED** [review]
This breaks `mdx`.

**Comment by marsam** [comment]
dune 2.8.1 was released https://github.com/ocaml/dune/releases/tag/2.8.1

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109325` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>dune-release: log was empty</li>
  </ul>
</details>
<details>
<summary>
39 packages built:
</summary>
<ul>
<li>acgtk</li>
<li>beluga</li>
<li>dot-merlin-reader</li>
<li>dune_2</li>
<li>eff</li>
<li>flow</li>
<li>framac</li>
<li>fstar</li>
<li>google-drive-ocamlfuse</li>
<li>jackline</li>
<li>libbap</li>
<li>libguestfs</li>
<li>liquidsoap</li>
<li>obelisk</li>
<li>ocaml-crunch</li>
<li>ocaml-top</li>
<li>ocamlformat (ocamlformat_0_15_0)</li>
<li>ocamlformat_0_11_0</li>
<li>ocamlformat_0_12</li>
<li>ocamlformat_0_13_0</li>
<li>ocamlformat_0_14_0</li>
<li>ocamlformat_0_14_1</li>
<li>ocamlformat_0_14_2</li>
<li>ocamlformat_0_14_3</li>
<li>opa</li>
<li>orpie</li>
<li>patdiff</li>
<li>python37Packages.bap</li>
<li>python37Packages.guestfs</li>
<li>python38Packages.bap</li>
<li>python38Packages.guestfs</li>
<li>python39Packages.bap</li>
<li>python39Packages.guestfs</li>
<li>reason</li>
<li>satysfi</li>
<li>stog</li>
<li>vagrant</li>
<li>virt-top</li>
<li>why3</li>
</ul>
</details>

**Comment by anmonteiro** [comment]
@vbgl the mdx failure looks like a broken formatter test, not an actual failure.

**Comment by anmonteiro** [comment]
@vbgl  Upgraded mdx too which now builds correctly

**Comment by vbgl** [comment]
Thanks. Merged the `mdx` update as 3ffc05e1dba2874cbb30b1ebc0e3dfe531e06249.

**Pull request merged**